World/Environment/Rain/Rain.gd

```diff
@@ -0,0 +1,113 @@
+@tool
+extends Node3D
+
+#
+@export var enabled := false:
+	set(val):
+		enabled = val
+		if get_child_count(): set_enabled(val)
+# From 0 (invisible rain drops) to 0.5 
+@export var drop_size := 0.3 :
+	set(val):
+		drop_size = val
+		if get_child_count(): set_drop_size(val)
+# From 0 (no rain) to 1 (very dense rain)
+@export var density := 0.0 :
+	set(val):
+		density = val
+		if get_child_count(): set_rain_density(val)
+# Cardinal direction of wind as unit vector and wind speed
+@export var wind_direction := Vector3(0, 0, 0) :
+	set(val):
+		wind_direction = val
+		if get_child_count(): set_wind(val, wind_speed)
+@export var wind_speed := 0.0 :
+	set(val):
+		wind_speed = val
+		if get_child_count(): set_wind(wind_direction, val)
+# Determines the extent of
+#	the individual rain drops/splashes/ripples
+# 	the start sheets (grouped texture of rain)
+@export var detailed_radius := 30 :
+	set(val):
+		detailed_radius = val
+		if get_child_count(): set_extents(val, coarse_radius, distance_radius)
+# Determines the extent of sheets
+@export var coarse_radius := 40 :
+	set(val):
+		coarse_radius = val
+		if get_child_count(): set_extents(detailed_radius, val, distance_radius)
+# Determines the extent of distant fog
+@export var distance_radius := 100 :
+	set(val):
+		distance_radius = val
+		if get_child_count(): set_extents(detailed_radius, coarse_radius, val)
+
+var heavy_rain_threshold := 0.5
```

Comment:
Also add a comment describing what this means

Reply:
[376c78c](https://github.com/boku-ilen/landscapelab/pull/326/commits/376c78c4101ae4aa9343e87ea69a3aed270435fb)